RELEASING/changelog.py

```diff
@@ -0,0 +1,254 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import os
+import re
+from dataclasses import dataclass
+from time import sleep
+from typing import Any, Dict, List, Optional, Union
+from urllib import request
+from urllib.error import HTTPError
+
+import click
+
+try:
+    import click
+except ModuleNotFoundError:
+    exit("Click is a required dependency for this script")
+
+
+@dataclass
+class GitLog:
+    """
+    Represents a git log entry
+    """
+
+    sha: str
+    author: str
+    time: str
+    message: str
+    pr_number: Union[int, None] = None
+
+    def __eq__(self, other: object) -> bool:
+        """ A log entry is considered equal if it has the same PR number """
+        if isinstance(other, self.__class__):
+            return other.pr_number == self.pr_number
+        return False
+
+    def __repr__(self):
+        return f"[{self.pr_number}]: {self.message} {self.time} {self.author}"
+
+
+class GitChangeLog:
+    """
+    Helper class to output a list of logs entries on a superset changelog format
+
+    We want to map a git author to a github login, for that we call github's API
+    """
+
+    def __init__(self, logs: List[GitLog]) -> None:
+        self._logs = logs
+        self._github_login_cache: Dict[str, Optional[str]] = {}
+        self._wait = 10
+
+    def _wait_github_rate_limit(self) -> None:
+        """
+        Waits for available rate limit slots on the github API
+        """
+        while True:
+            rate_limit_payload = self._fetch_github_rate_limit()
+            if rate_limit_payload["rate"]["remaining"] > 1:
+                break
+            print(f".", end="", flush=True)
```

Comment:
Nit:
```suggestion
            print(".", end="", flush=True)
```
We should probably also add a `print()` after the `while` to not continue the subsequent outputs from the same line.